Managers/FrameMan.cpp

```diff
@@ -1,2651 +1,1391 @@
-{
-	if (!g_SceneMan.SceneWrapsX() && !g_SceneMan.SceneWrapsY()) 
-	{ 
-		Vector drawStart = m_Start - targetPos; 
-		Vector drawEnd = m_End - targetPos; 
-		 
-		rectfill(pDrawScreen, drawStart.m_X, drawStart.m_Y, drawEnd.m_X, drawEnd.m_Y, m_Color);
-	} else {
-		Vector drawStartLeft;
-		Vector drawEndLeft;
-
-		Vector drawStartRight;
-		Vector drawEndRight;
-
-		TranslateCoordinates(targetPos, m_Start, drawStartLeft, drawStartRight);
-		TranslateCoordinates(targetPos, m_End, drawEndLeft, drawEndRight);
-
-		rectfill(pDrawScreen, drawStartLeft.m_X, drawStartLeft.m_Y, drawEndLeft.m_X, drawEndLeft.m_Y, m_Color);
-		rectfill(pDrawScreen, drawStartRight.m_X, drawStartRight.m_Y, drawEndRight.m_X, drawEndRight.m_Y, m_Color);
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void FrameMan::Update() {
+		// Update all the performance counters.
+		g_PerformanceMan.Update();
+
+		// Remove all scheduled primitives, those will be re-added by updates from other entities.
+		g_PrimitiveMan.ClearPrimitivesList();
 	}
-}
 
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
-void FrameMan::CirclePrimitive::Draw(BITMAP *pDrawScreen, Vector targetPos) 
-{
-	if (!g_SceneMan.SceneWrapsX() && !g_SceneMan.SceneWrapsY()) 
-	{ 
-		Vector drawStart = m_Start - targetPos; 
-		 
-		circle(pDrawScreen, drawStart.m_X, drawStart.m_Y, m_Radius, m_Color);
-	} else {
-		Vector drawStartLeft;
-		Vector drawStartRight;
+	void FrameMan::Draw() {
+		// Count how many split screens we'll need
+		int screenCount = (m_HSplit ? 2 : 1) * (m_VSplit ? 2 : 1);
 
-		TranslateCoordinates(targetPos, m_Start, drawStartLeft, drawStartRight);
+		RTEAssert(screenCount <= 1 || m_pPlayerScreen, "Splitscreen surface not ready when needed!");
 
-		circle(pDrawScreen, drawStartLeft.m_X, drawStartLeft.m_Y, m_Radius, m_Color);
-		circle(pDrawScreen, drawStartRight.m_X, drawStartRight.m_Y, m_Radius, m_Color);
-	}
-}
+		// Clear out the post processing screen effects list
+		g_PostProcessMan.Reset();
 
+		// These accumulate the effects for each player's screen area, and are then transferred to the above lists with the player screen offset applied
+		list<PostEffect> screenRelativeEffects;
+		list<Box> screenRelativeGlowBoxes;
 
-void FrameMan::CircleFillPrimitive::Draw(BITMAP *pDrawScreen, Vector targetPos) 
-{
-	if (!g_SceneMan.SceneWrapsX() && !g_SceneMan.SceneWrapsY()) 
-	{ 
-		Vector drawStart = m_Start - targetPos; 
-		 
-		circlefill(pDrawScreen, drawStart.m_X, drawStart.m_Y, m_Radius, m_Color);
-	} else {
-		Vector drawStartLeft;
-		Vector drawStartRight;
+		// Handy handle
+		Activity *pActivity = g_ActivityMan.GetActivity();
 
-		TranslateCoordinates(targetPos, m_Start, drawStartLeft, drawStartRight);
+		for (int whichScreen = 0; whichScreen < screenCount; ++whichScreen) {
+			// Screen Update
```

Comment:
Meaningless junk comment (seriously, I don't know what point this is trying to convey).